app/components/widgets/settings/text_setting.js

```diff
@@ -0,0 +1,228 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import {
+    View,
+    Text,
+    TextInput,
+    Platform,
+} from 'react-native';
+
+import FormattedText from 'app/components/formatted_text';
+import {changeOpacity, makeStyleSheetFromTheme} from 'app/utils/theme';
+
+export default class TextSetting extends PureComponent {
+    static propTypes = {
+        id: PropTypes.string.isRequired,
+        label: PropTypes.oneOfType([
+            PropTypes.shape({
+                id: PropTypes.string.isRequired,
+                defaultMessage: PropTypes.string.isRequired,
+            }),
+            PropTypes.string,
+        ]),
+        placeholder: PropTypes.string,
```

Comment:
there are multiple props here that are being passed to the TextInput I wonder why not using something like `...TextInput.propTypes` instead of setting each one individually?

e.g: https://github.com/mattermost/mattermost-mobile/blob/master/app/components/text_input_with_localized_placeholder.js#L11

Reply:
I'd prefer to be explicit so that we're not accidentally passing any props that we shouldn't be